datamonster_api/lib/datamonster.py

```diff
@@ -247,3 +255,272 @@ def _avro_to_df(self, avro_buffer):
+            things you can filter for.
+
+            EXAMPLE
+
+            Assuming `dm` is a DataMonster object, and given this datasource and company::
+
+                datasource = next(dm.get_datasources(query='1010data Credit Sales Index'))
+                the_gap = dm.get_company_by_ticker('GPS')
+
+            this call to `get_dimensions_for_datasource`::
+
+                dimset = dm.get_dimensions_for_datasource(
+                                datasource,
+                                filters={'section_pk': the_gap.pk,
+                                         'category': 'Banana Republic'})
+
+            returns an iterable, `dimset`, to a collection with just one dimensions dict.
+            Assuming `from pprint import pprint`, the following loop::
+
+                for dim in dimset:
+                    pprint(dim)
+
+            prettyprints the single dimension dict::
+
+                {'max_date': '2019-06-21',
+                 'min_date': '2014-01-01',
+                 'row_count': 1998,s
+                 'split_combination': {'category': 'Banana Republic',
+                                       'country': 'US',
+                                       'section_pk': 707}}]
+
+        :raises: DataMonsterError if `filters` is not a dict or is not JSON-serializable.
+            Re-raises `DataMonsterError` if self.client.get() raises that.
+        """
+        self._check_param(datasource=datasource)
+
+        params = {}
+        if filters:
+            params['filters'] = self.to_json_checked(filters)
+
+        url = self._get_dimensions_path(uuid=datasource.id)
+        if params:
+            url = ''.join([url, '?', six.moves.urllib.parse.urlencode(params)])
+
+        # Let any DataMonsterError from self.client.get() happen -- we don't occlude them
+        return DimensionSet(url, self, _convert_pks_to_tickers=_convert_pks_to_tickers)
+
+    def _convert_section_pks_to_tickers(self, dimension, pk2ticker_memos):
+        """
+        :param dimension: a dimension dict, with a key 'split_combination'.
+        :param pk2ticker_memos: (dict) maps `section_pk` => `ticker`
+            `section_pk's already in this dict will use the tickers already looked up and saved;
+            new `section_pk`s will have their tickers saved here
+
+        :return: `None`
+        Mutates the dict `dimension:
+            if 'section_pk' in `dimension['split_combination']`, its value::
+
+                dimension['split_combination']['section_pk"]
+
+            is a list of `section_pk`s (though we accommodate a single pk or `None`);
+            we replace each pk with
+
+            `self._pk_to_ticker(pk)`
+                  if ticker is not `None`,
+
+            name of company with key `pk`
+               if ticker is `None`
+        """
+        combo = dimension['split_combination']
+        if 'section_pk' in combo:
+            value = combo.pop('section_pk')
```

Comment:
Also, chatted with Elizabeth a but this morning and we should probably not pop this, having the section_pk available along side the ticker is useful in the cases where there is no ticker.

Reply:
Ah, sure -- add `'ticker'` item but don't remove `'section_pk'`

Reply:
(done)